### PR TITLE
Support peer task data multiplex

### PR DESCRIPTION
**Pull request by jim3ma**
## Related Issue

#348 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation Update (if none of the other choices apply)

## Checklist:

- [x] Code compiles correctly.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by gaius-qi — DISMISSED** [review]

File: client/daemon/storage/storage_manager.go [hunk 1]
```diff
@@ -143,6 +150,9 @@ func NewStorageManager(storeStrategy config.StoreStrategy, opt *config.StorageOp
 		tasks:         &sync.Map{},
 		dataPathStat:  stat.Sys().(*syscall.Stat_t),
 		gcCallback:    gcCallback,
+
+		indexRWMutex:       sync.RWMutex{},
```
Comment (244372610) [inline client/daemon/storage/storage_manager.go:154]: image

Comment (244372610) [reply client/daemon/storage/storage_manager.go:154]: No need to initialize

File: client/daemon/peer/peertask_file_test.go [hunk 2]
```diff
@@ -103,7 +103,7 @@ func TestFilePeerTask_BackSource_WithContentLength(t *testing.T) {
 			Url:      "http://localhost/test/data",
 			Filter:   "",
 			BizId:    "d7y-test",
-			UrlMata:  nil,
+			UrlMeta:  nil,
```
Comment (244372610) [inline client/daemon/peer/peertask_file_test.go:106]: image

Comment (244372610) [reply client/daemon/peer/peertask_file_test.go:106]: No initialization is required

File: client/daemon/peer/peertask_file_test.go [hunk 3]
```diff
@@ -214,7 +214,7 @@ func TestFilePeerTask_BackSource_WithoutContentLength(t *testing.T) {
...
```
Comment (244372610) [inline client/daemon/peer/peertask_file_test.go:217]: same as above

File: client/daemon/peer/peertask_manager_test.go [hunk 4]
```diff
@@ -206,7 +206,7 @@ func TestPeerTaskManager_StartFilePeerTask(t *testing.T) {
 			Url:      "http://localhost/test/data",
 			Filter:   "",
 			BizId:    "d7y-test",
-			UrlMata:  nil,
+			UrlMeta:  nil,
```
Comment (244372610) [inline client/daemon/peer/peertask_manager_test.go:209]: same as above

File: client/daemon/peer/peertask_manager_test.go [hunk 5]
```diff
@@ -281,7 +281,7 @@ func TestPeerTaskManager_StartStreamPeerTask(t *testing.T) {
 		Url:      "http://localhost/test/data",
 		Filter:   "",
 		BizId:    "d7y-test",
-		UrlMata:  nil,
+		UrlMeta:  nil,
```
Comment (244372610) [inline client/daemon/peer/peertask_manager_test.go:284]: same as above

File: client/daemon/peer/peertask_reuse.go [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+		},
+		TaskID:          reuse.TaskID,
+		PeerID:          request.PeerId,
+		ContentLength:   reuse.ContentLength,
+		CompletedLength: reuse.ContentLength,
+		PeerTaskDone:    true,
+		DoneCallback:    func() {},
+	}
+
+	progressCh := make(chan *FilePeerTaskProgress, 1)
```
Comment (244372610) [inline client/daemon/peer/peertask_reuse.go:84]: If chain is defined outside the method and only returns pg, is it better to be unified outside the method

File: client/daemon/peer/peertask_reuse.go [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+	var cost = time.Now().Sub(start).Milliseconds()
+	log.Infof("reuse file peer task done, cost: %dms", cost)
+
+	pg := &FilePeerTaskProgress{
+		State: &ProgressState{
+			Success: true,
+			Code:    dfcodes.Success,
+			Msg:     "Success",
+		},
+		TaskID:          reuse.TaskID,
```
Comment (244372610) [inline client/daemon/peer/peertask_reuse.go:76]: replace with request.TaskID, since they're the same,

File: client/daemon/peer/peertask_reuse.go [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+	log := logger.With("peer", request.PeerId, "task", reuse.TaskID, "component", "reuseStreamPeerTask")
+	ctx, span := tracer.Start(ctx, config.SpanStreamPeerTask, trace.WithSpanKind(trace.SpanKindClient))
+	span.SetAttributes(config.AttributePeerHost.String(ptm.host.Uuid))
+	span.SetAttributes(semconv.NetHostIPKey.String(ptm.host.Ip))
+	span.SetAttributes(config.AttributePeerID.String(request.PeerId))
+	span.SetAttributes(semconv.HTTPURLKey.String(request.Url))
+
+	rc, err := ptm.storageManager.ReadAllPieces(ctx, &reuse.PeerTaskMetaData)
+	if err != nil {
+		return nil, nil, false
```
Comment (244372610) [inline client/daemon/peer/peertask_reuse.go:105]: replace bool with error ？

**Review by 244372610 — DISMISSED** [review]

File: client/daemon/peer/peertask_reuse.go (hunk 7 above)
Comment (jim3ma) [reply client/daemon/peer/peertask_reuse.go:76]: There is no `TaskID` in request.

File: client/daemon/peer/peertask_reuse.go (hunk 6 above)
Comment (jim3ma) [reply client/daemon/peer/peertask_reuse.go:84]: There is no channel outside, so we create one.

File: client/daemon/peer/peertask_reuse.go (hunk 8 above)
Comment (jim3ma) [reply client/daemon/peer/peertask_reuse.go:105]: When reuse failed, we will download from p2p, did not care the error.
BTW, we just record the error in span

**Review by 244372610 — APPROVED** [review]

**Review by gaius-qi — APPROVED** [review]

**Review by 244372610 — APPROVED** [review]

**Pull request merged**